**Table of contents**<a id='toc0_'></a>    
- 1. [Importing Data](#toc1_)    
- 2. [Describing Data](#toc2_)    

<!-- vscode-jupyter-toc-config
	numbering=true
	anchor=true
	flat=false
	minLevel=1
	maxLevel=4
	/vscode-jupyter-toc-config -->
<!-- THIS CELL WILL BE REPLACED ON TOC UPDATE. DO NOT WRITE YOUR TEXT IN THIS CELL -->

# 1. <a id='toc1_'></a>[Importing Data](#toc0_)

In [1]:
#Importing Libraries
import pandas as pd
import numpy as np
import os

In [2]:
Path = r"D:\Data Analysis\01-08-2025 Instacart Basket Analysis"

In [3]:
vars_list = ['order_id',	'user_id',	'order_number',	'order_dow',	'order_hour_of_day',	'days_since_prior_order']

In [ ]:
df = pd.read_csv(os.path.join(Path, 'Data', 'Original Data', 'orders.csv'), usecols = vars_list)

In [5]:
df.head()

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2398795        1             2          3                  7   
2    473747        1             3          3                 12   
3   2254736        1             4          4                  7   
4    431534        1             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [12]:
df_prods = pd.read_csv(os.path.join(Path, 'Data', 'Original Data', 'products.csv'), usecols=['product_name',	'aisle_id',	'department_id',	'prices'])

In [13]:
df_prods.head(20)

product_name  aisle_id  \
0                          Chocolate Sandwich Cookies        61   
1                                    All-Seasons Salt       104   
2                Robust Golden Unsweetened Oolong Tea        94   
3   Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4                           Green Chile Anytime Sauce         5   
5                                        Dry Nose Oil        11   
6                      Pure Coconut Water With Orange        98   
7                   Cut Russet Potatoes Steam N' Mash       116   
8                   Light Strawberry Blueberry Yogurt       120   
9      Sparkling Orange Juice & Prickly Pear Beverage       115   
10                                  Peach Mango Juice        31   
11                         Chocolate Fudge Layer Cake       119   
12                                  Saline Nasal Mist        11   
13                     Fresh Scent Dishwasher Cleaner        74   
14                           Overnight Diapers Size 6        56   
15                      Mint Chocolate Flavored Syrup       103   
16                                  Rendered Duck Fat        35   
17                Pizza for One Suprema  Frozen Pizza        79   
18   Gluten Free Quinoa Three Cheese & Mushroom Blend        63   
19     Pomegranate Cranberry & Aloe Vera Enrich Drink        98   

    department_id  prices  
0              19     5.8  
1              13     9.3  
2               7     4.5  
3               1    10.5  
4              13     4.3  
5              11     2.6  
6               7     4.4  
7               1     1.1  
8              16     7.0  
9               7     8.4  
10              7     2.8  
11              1     9.4  
12             11     1.1  
13             17     6.5  
14             18    11.2  
15             19     5.2  
16             12    17.1  
17              1    10.6  
18              9    12.0  
19              7     6.0

In [14]:
df_prods.tail(35)

product_name  aisle_id  \
49658                   Teriyaki Sauce, Sesame, Original         5   
49659                                        Apple Cider        98   
49660                         Masada Kosher Pocket Bread       128   
49661                                  Cabernet Tomatoes        83   
49662                         Brie with Herbs Foil Wedge         2   
49663                            Organic Creamed Coconut        17   
49664                        Professionals Sleek Shampoo        22   
49665                                              Porto       134   
49666                       Bacon Cheddar Pretzel Pieces       107   
49667  Ultra Protein Power Crunch Peanut Butter N' Ho...        57   
49668                     Lemon Cayenne Drinking Vinegar       100   
49669      Super Dark Coconut Ash & Banana Chocolate Bar        45   
49670                      Ginger Snaps Snacking Cookies        61   
49671           Enchilada with Spanish Rice & Beans Meal        38   
49672                     Apple Cinnamon Scented Candles       101   
49673                                   K Cup Dark Blend       100   
49674                                Beef Summer Sausage       106   
49675                               Milk Chocolate Drops        45   
49676                             Cafe Mocha K-Cup Packs        26   
49677              Stone Baked Multi Grain Artisan Rolls       129   
49678            Frozen Greek Yogurt Bars Chocolate Chip        37   
49679                Cinnamon Dolce Keurig Brewed K Cups        26   
49680                             Ultra Red Energy Drink        64   
49681                    Thick & Chunky Sloppy Joe Sauce        59   
49682                    Large Chicken & Cheese Taquitos       129   
49683                            Famous Chocolate Wafers        61   
49684                 All Natural Creamy Caesar Dressing        89   
49685            Spaghetti with Meatballs and Sauce Meal        38   
49686                                 California Limeade        98   
49687                                     Cucumber Kirby        83   
49688          Vodka, Triple Distilled, Twist of Vanilla       124   
49689                 En Croute Roast Hazelnut Cranberry        42   
49690                                   Artisan Baguette       112   
49691         Smartblend Healthy Metabolism Dry Cat Food        41   
49692                             Fresh Foaming Cleanser        73   

       department_id  prices  
49658             13     4.0  
49659              7    10.7  
49660              3     7.1  
49661              4     8.3  
49662             16     3.9  
49663             13     3.1  
49664             11     6.7  
49665              5     8.2  
49666             19     3.6  
49667             14    10.2  
49668             21    13.7  
49669             19     6.9  
49670             19     5.2  
49671              1     6.6  
49672             17     5.6  
49673             21     4.7  
49674             12    19.2  
49675             19     3.0  
49676              7     6.5  
49677              1     5.6  
49678              1    11.1  
49679              7    14.0  
49680              7    14.5  
49681             15     8.9  
49682              1     3.4  
49683             19     6.0  
49684             13     4.9  
49685              1     6.9  
49686              7     4.3  
49687              4    13.2  
49688              5     5.3  
49689              1     3.1  
49690              3     7.8  
49691              8     4.7  
49692             11    13.5

# 2. <a id='toc2_'></a>[Describing Data](#toc0_)

In [15]:
df_prods.columns

Index(['product_name', 'aisle_id', 'department_id', 'prices'], dtype='object')

In [16]:
df_prods.shape

(49693, 4)

In [17]:
df_prods.describe()

aisle_id  department_id        prices
count  49693.000000   49693.000000  49693.000000
mean      67.770249      11.728433      9.994136
std       38.316774       5.850282    453.519686
min        1.000000       1.000000      1.000000
25%       35.000000       7.000000      4.100000
50%       69.000000      13.000000      7.100000
75%      100.000000      17.000000     11.200000
max      134.000000      21.000000  99999.000000

##### Answer: The maximum value of the aisle_id column is 134

In [18]:
df_prods.dtypes

product_name      object
aisle_id           int64
department_id      int64
prices           float64
dtype: object

##### Answer: The data type of the department_id is integer 64 bit